xarray/core/utils.py

```diff
@@ -188,7 +189,7 @@ def is_scalar(value):
     return (
         getattr(value, 'ndim', None) == 0 or
         isinstance(value, (basestring, bytes_type)) or not
-        isinstance(value, Iterable))
+        isinstance(value, (Iterable, dask_array_type)))
```

Comment:
`dask_array_type` is actually a tuple, so this needs to be `(Iterable,) + dask_array_type`.

That said... this already tricky logic is getting even more convoluted. So this isn't my favorite fix, though I agree that an N-dimensional dask array should not be considered a scalar.

Reply:
What would be a preferable fix here? It may be worth going back and seeing what the numpy array track is in `as_variable`.